dom-weather-app/public/app.js

```diff
@@ -0,0 +1,91 @@
+const cityInput = document.querySelector('#city-input');
+const temperature = document.querySelector('.temperature');
+const dayOrNight = document.querySelector('.time');
+const conditions = document.querySelector('.conditions');
+const icon = document.querySelector('.weather-icon');
+const errorMessage = document.querySelector('#error');
+
+const apiKey = 'qGLMGbgG2tHUnaRGFoJIvHMul7t55Esr';
+
+
+let timeout = null;
+
+cityInput.addEventListener('keyup', () => {
+    clearTimeout(timeout);
+
+    timeout = setTimeout(function () {
+        getCity(cityInput.value);
+    }, 2000);
+})
+
+const getCity = async (inputValue) => {
+
+    const base = 'http://dataservice.accuweather.com/locations/v1/cities/search';
+    const query = `?apikey=${apiKey}&q=${inputValue}`;
+
+    const response = await fetch(base + query)
+    .catch()
+    const data = await response.json();
+
+    return getWeather(data[0].Key);
+}
+
+const getWeather = async (cityKey) => {
+    const base = 'http://dataservice.accuweather.com/forecasts/v1/daily/1day/';
+    const query = `${cityKey}?apikey=${apiKey}`;
+    
+    const response = await fetch(base + query);
+    const data = await response.json();
+
+    return updateUI(data);
+}
+
+async function updateUI(weatherObject) {
+    // update day/night background
+    dayNight(whatHourIsIt());
+    const dayTime = dayNight(whatHourIsIt());
```

Comment:
widzisz jakie te 2 linijki są teraz dziwne, dayNight nie dość że zwraca info Day/Night ale jeszcze w tle coś aktualizuje. Teraz jak gdzieś będziesz chciał sprawdzić czy jest dzień/noc to zawsze to będzie ustawiało w DOMie też.

Reply:
ja tam w ogóle namieszałem, bo doczytałem w samym API, i okazuje się, że jest request który zwraca prostą flagę isDayLight: true/false i na tej podstawie można to oprzeć a nie się babrać w godziny :/ podmieniłem endpoint na który robie requesta i jest o wiele czyściej